### ISSUE TITLE
[REQ] Use data from treadmill for elevation in recorded FIT GPX

### ISSUE THREAD

**Issue by samkenw** [cagnulein/qdomyos-zwift#1812]
**Is your feature request related to a problem? Please describe.**
Context: I created a gpx file with 0 elevation, as I wanted to control the elevation manually by feel. I created the gpx, so that strava would register my elevation, as it requires gps data to take elevation into account. I then ran on the treadmill and adjusted the incline manually. However, after the run, the track data had 0 elevation - obviously because the app uses the the elevation from the provided gpx file (I assume) and not the treadmill.

**Describe the solution you'd like**
1. It would be nice, if the track data recorded in the FIT file, would use the "real" elevation data from the Treadmill/ App instead of the data in the original gpx file. 
2. Perhaps it would also be possible to add an option: "virtual track" or similar, which generates a gpx track as you go along based on distance and elevation recorded from the treadmill (or bike)  for any run - this would mean that elevation data would always be available in strava (in addition to having a track displayed) without the need for doing the whole Zwift setup etc. 

**Describe alternatives you've considered**
Alternatives are obviously Zwift etc. - this would just be a simple alternative to that. 

**Additional context**
Perhaps I am overseeing a reason why this does not make sense?

**Comment by cagnulein** [comment]
mmm no actually it tracks the elevation from the treadmill directly. do you have the debug log and the fit file?

**Comment by samkenw** [comment]
Ah strange. I sadly did not have debugging enabled. So, the elevation in the app was shown correctly but I have viewed the raw data in the FIT file and it has no elevation..

**Comment by cagnulein** [comment]
ok next time share the debug log and I will check it, it should work as you mentioned

**Comment by yellowbarchetta** [comment]
When I run normally on the treadmill, using QZ to record, it does record the elevation.  Are you saying that in the use-case you describe, where QZ uses a GPX file first, that this causes QZ to stop recording elevation in the .fit file (or it causes it to over-ride the treadmill's data).?

**Comment by samkenw** [comment]
Yes, this is what I believe happened, but I could be wrong. In the generated Fit file, I had the elevation from the GPX (which was 0) and not the actual elevation I set manually on the treadmill. I suppose I am misusing the GPX feature in this case because actually the app would try to automatically set the incline to 0 - however, I disabled the adb function so that I could set the incline manually. But I will have to go back and test it again to verify.
